gtest_parallel_tests.py

```diff
@@ -300,5 +303,133 @@ def test_get_save_file_path_win32(self):
                          gtest_parallel.get_save_file_path())
 
 
+class TestSerializeTestCases(unittest.TestCase):
+  class TaskManagerMock(object):
+    def __init__(self, group_checker):
+      self.group_checker = group_checker
+      self.check_lock = threading.Lock()
+
+    def run_task(self, task):
+      test_group = task.test_name.split('.')[0]
+
+      with self.check_lock:
+        self.group_checker.add(test_group)
```

Comment:
Why not get rid of the GroupChecker class and instead have a variable that stores whether tests from the same test groups have run in parallel.
Also, I think you should check all tasks were actually run.